docs/Utility/Log.md

```diff
@@ -0,0 +1,55 @@
+# Log
+
+Log class let's you write logs into custom log files.
+
+#### With default `Cake\Log\Log` class:
+
+To write log data into custom file with default CakePHP `Cake\Log\Log` class feels like repeating ourselves.
+
+1. Configure FileLog Adapter:
+```php
+use Cake\Log\Log;
+
+Log::config('custom_file', [
+    'className' => 'File',
+    'path' => LOGS,
+    'levels' => ['debug'],
+    'file' => 'my_file.log',
+]);
+```
+
+2. Write logs into file:
+```php
+Log::write('debug', "Something didn't work!");
+```
+
+With above approach, we have multiple issues:
+- It surely logs data into `custom_file.log` file but also it will log into level specific `$level.log` file too, so we end up duplicating the log data.
+- When we have to write more logs into same file from different parts of the project, we have to copy-paste this code every time.
+
+Or you hack it with doing something like setting configurations in `bootstrap.php` and use scope to log data. But each time you start new project you have to remember to copy paste that config and use in your project in order to write data into custom log files.
+
+#### With `Tools\Utility\Log` class:
+
+You can directly pass data to log and filename to write the data into.
+
+##### Usage:
+
+```php
+use Tools\Utility\Log;
+
+Log::write("Something didn't work!", 'my_file');
```

Comment:
So from what I understand in the code this is only for logging to files, right?
Maybe that class should be called FileLog then.

The problem I see is for example in my case I use logging to Database (with DatabaseLog) plugin switching out the existing streams from file to db.

This one here would be different as it would force always File.

Reply:
Yes, seems more relevant to use `FileLog::write`. I've changed it, please review.